distributed/diagnostics/tests/test_scheduler_plugin.py

```diff
@@ -125,3 +125,43 @@ async def close(self):
 
     assert plugin.history == ["started", "closed"]
     assert plugin.scheduler is s
+
+
+@gen_cluster(client=True)
+async def test_register_scheduler_plugin(c, s, a, b):
+    class Dummy1(SchedulerPlugin):
+        def start(self, scheduler):
+            scheduler.foo = "bar"
+
+    assert not hasattr(s, "foo")
+    await c.register_scheduler_plugin(Dummy1)
+    assert s.foo == "bar"
+
+    class Dummy2(SchedulerPlugin):
+        def start(self, scheduler):
+            raise RuntimeError("raising in start method")
+
+    n_plugins = len(s.plugins)
+    with pytest.raises(RuntimeError, match="raising in start method"):
+        await c.register_scheduler_plugin(Dummy2)
+    # total number of plugins should be unchanged
+    assert n_plugins == len(s.plugins)
+
+
+@gen_cluster(client=True, config={"distributed.scheduler.pickle": False})
+async def test_register_scheduler_plugin_pickle_disabled(c, s, a, b):
+    class Dummy1(SchedulerPlugin):
+        def start(self, scheduler):
+            scheduler.foo = "bar"
+
+    raise_match = (
+        "Cannot register a scheduler plugin as the scheduler "
+        "has been explicitly disallowed from deserializing "
+        "arbitrary bytestrings using pickle via the "
+        "'distributed.scheduler.pickle' configuration setting."
+    )
+
+    n_plugins = len(s.plugins)
+    with pytest.raises(ValueError, match=raise_match):
+        await c.register_scheduler_plugin(Dummy1)
```

Comment:
Nitpick: Typically we don't explicitly check the full error message which is raised and instead check that one or more key phrases are contained within the error message. This tends to make tests more robust (though not always) to small changes in the message that is raised. 

For this specific test the main things we want to make sure is in the error message is that the scheduler is configured in a way such that it can't deserialize with pickle. To handle this, we might check that `disallowed from deserializing` and `distributed.scheduler.pickle` (the corresponding configuration setting) are included in the error message which is raised: 

```diff
diff --git a/distributed/diagnostics/tests/test_scheduler_plugin.py b/distributed/diagnostics/tests/test_scheduler_plugin.py
index e19c0a34..73d380af 100644
--- a/distributed/diagnostics/tests/test_scheduler_plugin.py
+++ b/distributed/diagnostics/tests/test_scheduler_plugin.py
@@ -154,14 +154,11 @@ async def test_register_scheduler_plugin_pickle_disabled(c, s, a, b):
         def start(self, scheduler):
             scheduler.foo = "bar"

-    raise_match = (
-        "Cannot register a scheduler plugin as the scheduler "
-        "has been explicitly disallowed from deserializing "
-        "arbitrary bytestrings using pickle via the "
-        "'distributed.scheduler.pickle' configuration setting."
-    )
-
     n_plugins = len(s.plugins)
-    with pytest.raises(ValueError, match=raise_match):
+    with pytest.raises(ValueError) as excinfo:
         await c.register_scheduler_plugin(Dummy1)
+    msg = str(excinfo.value)
+    assert "disallowed from deserializing" in msg
+    assert "distributed.scheduler.pickle" in msg
+
     assert n_plugins == len(s.plugins)
```

Reply:
thanks for explaining!